include/drivers/gpio.h

```diff
@@ -201,79 +201,6 @@ extern "C" {
 
 /** @} */
 
-/** Enable GPIO pin debounce.
- *
- * @note Drivers that do not support a debounce feature should ignore
- * this flag rather than rejecting the configuration with -ENOTSUP.
- */
-#define GPIO_INT_DEBOUNCE              (1U << 19)
```

Comment:
We cannot remove the macros, need to deprecate them first, e.g.:
```
#define GPIO_INT_DEBOUNCE              __DEPRECATED_MACRO (1U << 8)
```
The same holds true for drive strength flags.

Reply:
It's tricky to both deprecate them and make them SoC specific without overlapping.

Reply:
When going the `__DEPRECATED_MACRO` way we would need to make sure that the code compiles and the drivers work just as before. The only difference being the warning message generated by the compiler.

This is possible as long as we use the same bits in the SoC specific flags. I.e. in `include/drivers/gpio.h`
```
#define GPIO_DS_LOW_POS __DEPRECATED_MACRO 9
#define GPIO_DS_HIGH_POS __DEPRECATED_MACRO 11
```
and then in the SoC specific flags e.g. `include/dt-bindings/gpio/nordic-nrf-gpio.h`
```
#define NRF_GPIO_DS_LOW_POS 9
#define NRF_GPIO_DS_HIGH_POS 11
```
We need to keep the deprecated macros for two releases. When they are gone the SoC specific flags can be cleaned up, if required.

Reply:
@mnkp Thank you for the feedback. I modified this PR to do that.